src/goTypeDefinition.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import vscode = require('vscode');
+import cp = require('child_process');
+import path = require('path');
+import { byteOffsetAt, getBinPath, canonicalizeGOPATHPrefix, getFileArchive, killTree } from './util';
+import { promptForMissingTool, promptForUpdatingTool } from './goInstallTools';
+import { getToolsEnvVars } from './util';
+
+interface GuruDescribeOutput {
+	desc: string;
+	pos: string;
+	detail: string;
+	value: GuruDescribeValueOutput;
+}
+
+interface GuruDescribeValueOutput {
+	type: string;
+	value: string;
+	objpos: string;
+	typespos: GuruDefinitionOutput[];
+}
+
+interface GuruDefinitionOutput {
+	objpos: string;
+	desc: string;
+}
+
+export class GoTypeDefinitionProvider implements vscode.TypeDefinitionProvider {
+	provideTypeDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): vscode.ProviderResult<vscode.Definition> {
+		return new Promise<vscode.Location[]>((resolve, reject) => {
+			let goGuru = getBinPath('guru');
+			if (!path.isAbsolute(goGuru)) {
+				promptForMissingTool('guru');
+				return reject('Cannot find tool "guru" to find type definitions.');
+			}
+
+			let filename = canonicalizeGOPATHPrefix(document.fileName);
+			let offset = byteOffsetAt(document, position);
+			let env = getToolsEnvVars();
+			let buildTags = vscode.workspace.getConfiguration('go', document.uri)['buildTags'];
+			let args = buildTags ? ['-tags', buildTags] : [];
+			args.push('-json', '-modified', 'describe', `${filename}:#${offset.toString()}`);
+
+			let process = cp.execFile(goGuru, args, { env }, (err, stdout, stderr) => {
```

Comment:
To keep `guru implements` fast, we restrict the scope to current workspace in https://github.com/Microsoft/vscode-go/blob/0.7.0/src/goImplementations.ts#L32 which implements the `Go to Implementation` feature.

Would the same apply here as well?

Reply:
My measurement results says that scope doesn't affect describe's performance. I think describe looking for type definition from imports only - there is no need to analyze whole workspace and no need in scope.

Reply:
@ramya-rao-a I can try it on some big project from github and show measurements result. Should I?

Reply:
@hummerd Yes please :) If you can try on some big projects with and without the scope and then share the results, that would be great

Reply:
So I tried it on kubernetes. I tried:
```
# from go/src/k8s.io/kubernetes/pkg/kubectl
time guru -json describe apply.go:#3880
time guru -json -scope . describe apply.go:#3880
time guru -json -scope k8s.io/kubernetes/pkg/kubectl describe apply.go:#3880
```
All gives same results (on my machine mean real ~ 2.03s)

Reply:
Thanks @hummerd!